### DISCUSSION TITLE
LockIsolated value access with a timeout

### DISCUSSION THREAD

**Discussion by rhr-gc** [pointfreeco/swift-concurrency-extras#37] [Ideas]
I see that `LockIsolated` uses an `NSRecursiveLock` as the synchronization mechanism. I also notice that to get a value stored by `LockIsolated` the pattern is to use the `value` computed property. This calls into the `lock()` function of `NSRecursiveLock`. 

`NSLocking` provides a method `lock(before:)` that allows client code to supply a timeout parameter. I think this would be a useful addition to `LockIsolated`. In my specific case, there is a low-priority piece of data that would be nice to have when sending along a network request, but not necessary - and ultimately not worth blocking the network request. Being able to request a `LockIsolated` value with a timeout would be valuable.

**Comment by mbrandonw** [comment]
Hi @rhr-gc, this is definitely an interesting idea, but in the long run we plan on deprecating `LockIsolated` in favor of the new `Mutex` type in Swift 6. So we don't think we should be adding new features to this type. Luckily the type is simple enough that you could always maintain your own version of it.
